CMakeLists.txt

```diff
@@ -79,12 +83,12 @@ target_include_directories(ArborX INTERFACE
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/interpolation>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/spatial>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
-  $<INSTALL_INTERFACE:include/ArborX>
-  $<INSTALL_INTERFACE:include/ArborX/geometry>
-  $<INSTALL_INTERFACE:include/ArborX/cluster>
-  $<INSTALL_INTERFACE:include/ArborX/distributed>
-  $<INSTALL_INTERFACE:include/ArborX/interpolation>
-  $<INSTALL_INTERFACE:include/ArborX/spatial>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX/geometry>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX/cluster>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX/distributed>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX/interpolation>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ArborX/spatial>
```

Comment:
`CMAKE_INSTALL_INCLUDEDIR` is just `include` by default, after this PR a user could change it to something else.